### PR TITLE
Ignore cookies that fail decryption

### PR DESCRIPTION
**Pull request by matthewmcgarvey**

### CODE REVIEW

**Comment by jwoertink** [comment]
Tested this in the website, and it looks good from there, at least. We will need to find a way to test this with some existing project that utilizes session a bit more. But I like the direction better at least.

**Comment by stephendolan** [comment]
Happy to test on ReviewMy.email! I usually upgrade it when I'm going through the LuckyDiff release.

Should I wait until this is merged or point directly to this branch in a shard.override in prod?

**Comment by icy-arctic-fox** [comment]
Tested on a project I'm working on saw no issues. Was using Lucky 0.26.0 but overrode it with this branch.

I had a logged in session and had to log back in (was kicked out). Had no issues logging in and out multiple times.

**Comment by matthewmcgarvey** [comment]
This is where rails handles this error and probably how we should as well https://github.com/rails/rails/blob/a27c9eeddb19eaabf2e6d808152b18d5ca17736f/actionpack/lib/action_dispatch/middleware/cookies.rb#L671-L677

So any cookie we try to decrypt and fails we return nil. So we should move the error handling into CookieJar and out of Session

**Comment by robacarp** [comment]
I like this much better ❤️

File: spec/lucky/cookies/cookie_jar_spec.cr
Comment (matthewmcgarvey) [inline spec/lucky/cookies/cookie_jar_spec.cr:81]: Hopefully this spec will help in the future if we have to change cookies again

File: src/lucky/cookies/cookie_jar.cr
Comment (robacarp) [reply src/lucky/cookies/cookie_jar.cr:153]: I think the reason they're only rescuing specific errors is that they don't want the blanket catch-all. If something goes wrong in an adjacent library which _isn't_ a decryption failure, it would cause a false positive on this rescue. 

The risk is, some code changes in Base64 or lchop or something which raises an exception, the tests pass for some reason, and now no cookies are readable and the code is just assuming it's a crypto failure.

File: spec/lucky/cookies/cookie_jar_spec.cr
Comment (robacarp) [reply spec/lucky/cookies/cookie_jar_spec.cr:81]: I think this makes sense, but it's also going to be difficult to maintain this test if it needs to change for some reason because of the opaque cookie value. I know it's redundant because the crypto code exists in the other file, but does it make sense to put a few lines of explicit crypto in here which builds the cookie value instead of the raw base64 encoded value?

File: src/lucky/cookies/cookie_jar.cr
Comment (jwoertink) [reply src/lucky/cookies/cookie_jar.cr:153]: We could add a log here if we need. 

```
Lucky::Log.dexter.info { {message: "Cookie decrypt failed", error: e.message} }
```

**Review by jwoertink — APPROVED** [review]
I think this is a better direction. Feel free to merge when ready

**Pull request merged**
